packages/fxa-settings/src/pages/AccountRecoveryConfirmKey/index.tsx

```diff
@@ -0,0 +1,169 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Link, RouteComponentProps } from '@reach/router';
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { logViewEvent, usePageViewEvent } from '../../lib/metrics';
+import { useAlertBar } from '../../models';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useFtlMsgResolver } from '../../models/hooks';
+
+import { InputText } from '../../components/InputText';
+import CardHeader from '../../components/CardHeader';
+import WarningMessage from '../../components/WarningMessage';
+import ResetPasswordLinkExpired from '../../components/ResetPasswordLinkExpired';
+
+// --serviceName-- is the relying party
+
+export type AccountRecoveryConfirmKeyProps = {
+  serviceName?: string;
+  linkStatus: LinkStatus;
+};
+
+type FormData = {
+  recoveryKey: string;
+};
+
+type LinkStatus = 'expired' | 'valid';
```

Comment:
Does this not have a damaged link status?

Reply:
It doesn't look like the original content-server view for this component included a "damaged link" case. Could be good to add though?

Reply:
That's so curious...I guess it might be good to add it in?